runbot_send_email/models/runbot_build.py

```diff
@@ -0,0 +1,76 @@
+# coding: utf-8
+# © 2016 Vauxoo
+#   Coded by: lescobar@vauxoo.com
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+
+from openerp import api, models
+
+_logger = logging.getLogger(__name__)
+
+
+class RunbotBuild(models.Model):
+    _name = "runbot.build"
+    _inherit = ['runbot.build', 'mail.thread']
+
+    @api.multi
+    def action_send_email(self):
+        self.ensure_one()
+        ir_model_data = self.env['ir.model.data']
+        try:
+            template_id = ir_model_data.get_object_reference(
+                                                        'runbot_send_email',
+                                                        'runbot_send_notif')[1]
+        except ValueError:
+            template_id = False
+        try:
+            compose_form_id = ir_model_data.get_object_reference('mail',
+                                    'email_compose_message_wizard_form')[1]
+        except ValueError:
+            compose_form_id = False
+        ctx = dict()
+        ctx.update({
+            'default_model': 'runbot',
+            'default_res_id': self.ids[0],
+            'default_use_template': bool(template_id),
+            'default_template_id': template_id,
+            'default_composition_mode': 'comment',
+            'mark_so_as_sent': True
+        })
+        return {
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'mail.compose.message',
+            'views': [(compose_form_id, 'form')],
+            'view_id': compose_form_id,
+            'target': 'new',
+            'context': ctx,
+        }
+
+    @api.multi
+    def send_email(self):
+        name_build = self.dest
+        email_to = self.committer_email
+        partner_obj = self.env['res.partner']
+        partner_id = partner_obj.find_or_create(email_to)
+        partner = partner_obj.browse(partner_id)
+        if partner not in self.message_partner_ids:
+            self.message_subscribe([partner.id])
+        email_act = self.action_send_email()
+        if email_act and email_act.get('context'):
+            email_ctx = email_act['context']
+            self.with_context(email_ctx).message_post_with_template(
+                                                    email_ctx.get(
+                                                        'default_template_id'))
+            _logger.info('Sent email to: %s, Build: %s', email_to, name_build)
+        return True
+
+    @api.multi
+    def github_status(self):
+        super(RunbotBuild, self).github_status()
+        try:
+            self.send_email()
+        except BaseException, e:
```

Comment:
@lescobarvx try to use a less generic Exception:

i.e.; If no mail server the exception is 1, if no login correctly setted 2 if Other then you can manage correctly the try,

BUT BTW the message_post in odoo has its own try: then you do not need to wrap this method here,.